File: libraries/managed-exceptions-lib/src/python/managed_exceptions/argumemts/item_already_exists_exception.py
from http import HTTPStatus
from managed_exceptions.error_details import ErrorDetails
from managed_exceptions.managed_exception import ManagedException

class ItemAlreadyExistsException(ManagedException):
    def __init__(self, message: str, diagnostic_details: dict[str, str] = {}):
        super().__init__(ErrorDetails(
            status_code=HTTPStatus.CONFLICT,
            diagnostic_code="00409",
            diagnostic_details=diagnostic_details,
            message=message
        ))

File: services/Sphana/Libraries/Sphana.ManagedException/Arguments/ItemAlreadyExistsException.cs
using System.Net;

namespace Sphana.ManagedException.Arguments;

/// <summary>
/// Thrown when attempting to create an item that already exists.
/// </summary>
public class ItemAlreadyExistsException : ManagedException
{
    public ItemAlreadyExistsException(string message, Dictionary<string, string>? diagnosticDetails = null)
        : base(HttpStatusCode.Conflict, "ITEM_ALREADY_EXISTS", message, diagnosticDetails)
    {
    }
}